### ISSUE TITLE
Add device with same name fails

### ISSUE THREAD

**Issue by GubblGubbl** [netbox-community/netbox#10386]
### NetBox version

v.3.3.3

### Python version

3.10

### Steps to Reproduce

1. Create site a
2. Create location a on site a
3. Create location b on site a
4. Create device a on site a location a
5. Create device a on site a location b

### Expected Behavior

Device a will be created on both locations.

### Observed Behavior

Device a is created on location a.
Creating device a on location b results in "A device with this name already exists."

**Comment by dreng** [comment]
This might be considered a bug, but honestly you should definitely develop a proper naming convention that takes into account all sites and locations. You should never have duplicate names if you do it properly.

**Comment by GubblGubbl** [comment]
We discovered this behavior accidentally by importing devices not with live data :)

**Comment by kkthxbye-code** [comment]
As far as I know this is intended and as such is not a bug. If you want to suggest a change it must be raised as a feature request.

[From the documentation.](https://demo.netbox.dev/static/docs/models/dcim/device/)

> Device names must be unique within a site, unless the device has been assigned to a tenant. Devices may also be unnamed.

https://github.com/netbox-community/netbox/blob/1d033bd286f605dd4c500b3808cb496b347d4d3c/netbox/dcim/models/devices.py#L682-L691

https://github.com/netbox-community/netbox/blob/1d033bd286f605dd4c500b3808cb496b347d4d3c/netbox/dcim/models/devices.py#L652-L658

In short, uniqueness of devices is based on name, tenant and site. So you cannot have two devices with the same name on the same site with the same tenant. If you want location to be part of the uniqueness constraint it must be raised as a FR.
